invenio_rdm_records/records/systemfields/is_published.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record has draft check field.
+
+The HasDraftCheckField is used to check if an associated draft exists for a
+a record.
+"""
+
+from invenio_records.dictutils import dict_set
+from invenio_records.systemfields import SystemField
+from invenio_records_resources.records.systemfields import PIDStatusCheckField
+from sqlalchemy.orm.exc import NoResultFound
+
+
+class IsPublishedField(PIDStatusCheckField):
+    """Adds a `pre_dump` hook to pid status check field."""
+
+    def pre_dump(self, record, **kwargs):
+        """Called before a record is dumped in a secondary storage system."""
+        dict_set(
```

Comment:
I'm changing Invenio-Records so that the signature of pre_dump is:

```python
def pre_dump(record, data, dumper=None):
    pass
```

Also, there's a ``post_dump()`` (same signature).

This avoid modifying the record, because if you commit after having dumped, you'll have the property inside the record. See https://github.com/inveniosoftware/invenio-records/pull/249